toolkit/tools/pkg/imagecreatorlib/imagecreator.go

```diff
@@ -0,0 +1,217 @@
+	osRelease            string
+}
+
+func CreateImageWithConfigFile(buildDir string, configFile string,
+	rpmsSources []string,
+	toolsTar string,
+	outputImageFile string,
+	outputImageFormat string,
+) error {
+	var config imagecustomizerapi.Config
+	err := imagecustomizerapi.UnmarshalYamlFile(configFile, &config)
+	if err != nil {
+		return err
+	}
+
+	baseConfigPath, _ := filepath.Split(configFile)
+
+	absBaseConfigPath, err := filepath.Abs(baseConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path of config file directory:\n%w", err)
+	}
+	useBaseImageRpmRepos := false
+
+	err = createNewImage(buildDir, absBaseConfigPath, config, outputImageFile, rpmsSources, outputImageFile, outputImageFormat, useBaseImageRpmRepos, toolsTar)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createNewImage(buildDir string, baseConfigPath string, config imagecustomizerapi.Config, inputImageFile string,
+	rpmsSources []string, outputImageFile string, outputImageFormat string,
+	useBaseImageRpmRepos bool, toolsTar string,
+) error {
+	if toolsTar == "" {
+		return fmt.Errorf("tools tar file is required for image creation")
+	}
+
+	// TODO: Add validation for the config file wrt the imager config
+	err := imagecustomizerlib.ValidateConfig(baseConfigPath, &config, inputImageFile, rpmsSources, outputImageFile, outputImageFormat, useBaseImageRpmRepos, "", true)
+	if err != nil {
+		return err
+	}
+
+	imageCreatorParameters, err := createImageCreatorParameters(buildDir,
+		baseConfigPath, &config, rpmsSources,
+		outputImageFormat, outputImageFile, "", toolsTar)
+	if err != nil {
+		return fmt.Errorf("invalid parameters:\n%w", err)
+	}
+
+	err = imagecustomizerlib.CheckEnvironmentVars()
+	if err != nil {
+		return err
+	}
+
+	imagecustomizerlib.LogVersionsOfToolDeps()
+
+	// ensure build and output folders are created up front
+	err = os.MkdirAll(imageCreatorParameters.buildDirAbs, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	err = os.MkdirAll(imageCreatorParameters.outputImageDir, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	disks := imageCreatorParameters.config.Storage.Disks
+	diskConfig := disks[0]
+	installOSFunc := func(imageChroot *safechroot.Chroot) error {
+		return nil
+	}
+
+	logger.Log.Infof("Creating new image with parameters: %+v\n", imageCreatorParameters)
+
+	// TODO: Get the target OS from the config or command line argument
+	partIdToPartUuid, err := imagecustomizerlib.CreateNewImage(targetos.TargetOsAzureLinux3, imageCreatorParameters.rawImageFile, diskConfig, imageCreatorParameters.config.Storage.FileSystems,
+		imageCreatorParameters.buildDirAbs, setupRoot, installOSFunc)
+	if err != nil {
+		return err
+	}
+
+	logger.Log.Debugf("Part id to part uuid map %v\n", partIdToPartUuid)
+
+	logger.Log.Infof("Customizing image with UUID %s", imageCreatorParameters.imageUuidStr)
+	// print the output image file
+	logger.Log.Infof("Output image file: %s", imageCreatorParameters.outputImageFile)
```

Comment:
You already have a `"Writing: %s"` log message. So, you probably don't need to duplicate that information.